### PR TITLE
php.extensions: Updates to multiple extensions

### PR DESCRIPTION
**Pull request by etu**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126351` at 147c66e0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built successfully:</summary>
  <ul>
    <li>libcouchbase</li>
    <li>php74Extensions.couchbase</li>
    <li>php74Extensions.igbinary</li>
    <li>php74Extensions.mongodb</li>
    <li>php74Extensions.pcov</li>
    <li>php74Extensions.protobuf</li>
    <li>php74Extensions.redis</li>
    <li>php74Extensions.xdebug</li>
    <li>php80Extensions.couchbase</li>
    <li>php80Extensions.igbinary</li>
    <li>php80Extensions.mongodb</li>
    <li>php80Extensions.pcov</li>
    <li>php80Extensions.protobuf</li>
    <li>php80Extensions.redis</li>
    <li>php80Extensions.xdebug</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/build-support/build-pecl.nix:36:3:
```
   |
36 |   checkPhase = "NO_INTERACTON=yes make test";
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/php-packages/protobuf/default.nix:1:26:
```
  |
1 | { buildPecl, lib, pcre2, fetchpatch }:
  |                          ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/libcouchbase/default.nix:24:3:
```
   |
24 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/php-packages/couchbase/default.nix:22:5:
```
   |
22 |     (writeText "php-couchbase.patch" ''
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/php-packages/couchbase/default.nix [hunk 1]
```diff
@@ -10,37 +10,27 @@ buildPecl {
...
-      @@ -154,6 +154,8 @@ COUCHBASE_FILES=" \
-           igbinary_inc_path="$phpincludedir"
-         elif test -f "$phpincludedir/ext/igbinary/igbinary.h"; then
-           igbinary_inc_path="$phpincludedir"
-      +  elif test -f "${php.extensions.igbinary.dev}/include/ext/igbinary/igbinary.h"; then
-      +    igbinary_inc_path="${php.extensions.igbinary.dev}/include"
-         fi
-         if test "$igbinary_inc_path" = ""; then
-           AC_MSG_WARN([Cannot find igbinary.h])
     '')
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/couchbase/default.nix:34]: I don't think we should inline patches. Can we move this to an actual patchfile that uses substituteAll?

File: pkgs/development/php-packages/protobuf/default.nix [hunk 2]
```diff
@@ -3,28 +3,11 @@
...
-  version = "3.14.0";
-  sha256 = "1ldc4s28hq61cfg8l4c06pgicj0ng7k37f28a0dnnbs7xkr7cibd";
+  version = "3.17.2";
+  sha256 = "0i4npj4sl8ihkzxc6m3vv3nlqk952z9bfwnrk90a9yakw5gfhlz5";
 
   buildInputs = [ pcre2 ];
 
-  patches = [
-    # TODO: remove with next update
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/protobuf/default.nix:13]: Please remove the unused fetchpatch from inputs.

File: pkgs/development/php-packages/couchbase/default.nix (hunk 1 above)
Comment (etu) [reply pkgs/development/php-packages/couchbase/default.nix]: Fixed :)

**Review by risicle — APPROVED** [review]
LGTM, `nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Comment by etu** [comment]
Found more to update... :slightly_smiling_face:

**Comment by dasJ** [comment]
Have you checked if some of these need to be backported? Both for security updates and bugfixes

**Comment by etu** [comment]
@dasJ I'm thinking all minor and patch updates should be fine, so it's mostly... couchbase that should be ignored.

**Pull request merged**